main.cpp

```diff
@@ -493,19 +562,59 @@ struct DigitalWorkstation
         float sartTimeSelectionPoint = 0.0f;
         float endTimeSelectionPoint = 24.0f;
 
-        void loopSelection(bool loopSelcetion);
-        int tapTempoBpm (int numTap, float time);
+        void loopSelection(bool loopSelection);
+        float tapTempoBpm (int numTap, float time);
         void moveCursorPos (int beat, int mesure);
     };
 
-    //loop audio and midi sections
     void loopAudioSection (float loopIn, float loopOut);
-    //record audio and midi performance 
     void recordData (int input, bool isMidi);
-    //mix multiple recordings and tracks into a final produced
     void bounce (int tracks, float startingPoint, float endingPoint);
 };
 
+void DigitalWorkstation::loopAudioSection(float loopIn, float loopOut)
+{
+    loopIn -= loopOut;
+}
+
+void DigitalWorkstation::recordData(int input, bool isMidi){
+    if (isMidi)
+    {
+        input = 1;
+    }
+    else
+    {
+        input = 0;
+    }
+}
+
+void DigitalWorkstation::bounce(int tracks, float startingPoint, float endingPoint)
+{
+    tracks = tracks * 2;
+    startingPoint += endingPoint;
+}
+
+void DigitalWorkstation::TransportBar::loopSelection(bool loopSelection)
+{
+    if (loopSelection)
+    {
+        loopSelection = false;
+    }
+}
+
+float DigitalWorkstation::TransportBar::tapTempoBpm(int numTap, float time)
+{
+    float newBpm;
+    newBpm = numTap/time;
+
+    return newBpm;
```

Comment:
just write `return numTap / time;`  
no need for the local variable.
- [ ] revise